### ISSUE TITLE
[FYI] boost spirit x3 will move to c++17

### ISSUE THREAD

**Issue by mathisloge** [mapnik/mapnik#4310]
https://www.boost.org/doc/libs/develop/libs/spirit/doc/x3/html/index.html

at the bottom it said that boost spirit x3 will move to c++17 with version 1.81

**Comment by artemp** [comment]
@mathisloge thanks for tracking this! We can still allow `c++14` builds with `BOOST_VERSION < 1_81` and bump up requirement to `-std=c++17` by default?

**Comment by mathisloge** [comment]
@artemp yep. I think we can default to c++17 in CMake now but support c++14 with `BOOST_VERSION < 1_81`

So one have to set `CMAKE_CXX_STANDARD=14` in the cmake configure step

**Comment by artemp** [comment]
@mathisloge sounds good!
